Extension/src/LanguageServer/configurations.ts

```diff
@@ -675,15 +692,170 @@ export class CppProperties {
+            let forcedIncludeStart: number = curText.search(/\s*\"forcedInclude\"\s*:\s*\[/);
+            let forcedeIncludeEnd: number = forcedIncludeStart === -1 ? -1 : curText.indexOf("]", forcedIncludeStart);
+            let compileCommandsStart: number = curText.search(/\s*\"compileCommands\"\s*:\s*\"/);
+            let compileCommandsEnd: number = compileCommandsStart === -1 ? -1 : curText.indexOf('"', curText.indexOf('"', curText.indexOf(":", compileCommandsStart)) + 1);
+
+            if (this.prevSquiggleMetrics[this.CurrentConfiguration.name] === undefined) {
+                this.prevSquiggleMetrics[this.CurrentConfiguration.name] = { PathNonExistent: 0, PathNotAFile: 0, PathNotADirectory: 0 };
+            }
+            let newSquiggleMetrics: { [key: string]: number } = { PathNonExistent: 0, PathNotAFile: 0, PathNotADirectory: 0 };
+
+            for (let curPath of paths) {
+                let resolvedPath: string = curPath.substr(1, curPath.length - 2);
+                // Resolve special path cases.
+                if (resolvedPath === "${default}") {
+                    // TODO: Add squiggles for when the C_Cpp.default.* paths are invalid.
+                    continue;
+                }
+                resolvedPath = util.resolveVariables(resolvedPath, this.ExtendedEnvironment);
+                if (resolvedPath.includes("${workspaceFolder}")) {
+                    resolvedPath = resolvedPath.replace("${workspaceFolder}", this.rootUri.fsPath);
+                }
+                if (resolvedPath.includes("${workspaceRoot}")) {
+                    resolvedPath = resolvedPath.replace("${workspaceRoot}", this.rootUri.fsPath);
+                }
+                if (resolvedPath.includes("${vcpkgRoot}")) {
+                    resolvedPath = resolvedPath.replace("${vcpkgRoot}", util.getVcpkgRoot());
+                }
+                if (resolvedPath.includes("*")) {
+                    resolvedPath = resolvedPath.replace(/\*/g, "");
+                }
+
+                // Handle WSL paths.
+                if (resolvedPath.startsWith("/") && os.platform() === 'win32') {
+                    const mntStr: string = "/mnt/";
+                    if (resolvedPath.length > "/mnt/c/".length && resolvedPath.substr(0, mntStr.length) === mntStr) {
+                        resolvedPath = resolvedPath.substr(mntStr.length);
+                        resolvedPath = resolvedPath.substr(0, 1) + ":" + resolvedPath.substr(1);
+                    } else if (this.rootfs && this.rootfs.length > 0) {
+                        resolvedPath = this.rootfs + resolvedPath.substr(1);
+                        resolvedPath = resolvedPath.replace(/\//g, path.sep);
+                        // TODO: Handle WSL symlinks.
+                    }
+                }
+
+                let pathExists: boolean = true;
+                if (!fs.existsSync(resolvedPath)) {
+                    // Check again for a relative path.
```

Comment:
I thought we didn't support relative paths at all

Reply:
What do you mean? What didn't support relative paths? The language server supports relative paths. Do you see a problem with the squiggle support for this?

Reply:
`includePath`, for example doesn't support relative paths.

Reply:
Okay, you're right. I tested it and it seemed to work, but it was actually using the file-relative path and not the workspaceFolder relative path.

Reply:
Wait -- I'm seeing relative include paths work sometimes. Might be some bug.

Reply:
Relative inclue paths seem to all work for me. It's better to not show squiggles if there arer working cases.